files/scripts/swss.sh

```diff
@@ -107,6 +107,92 @@ function clean_up_tables()
     end" 0
 }
 
+# This function cleans up the chassis db table entries created ONLY by this asic
+# This is used to do the clean up operation when the line card / asic reboots
+# When the asic/lc is RE-booting, the chassis db server is supposed to be running 
+# in the supervisor.  So the clean up is done when only the chassis db connectable. 
+# Otherwise no need to do the clean up since both the supervisor and line card may be 
+# rebooting (the whole chassis scenario)
+# The clean up operation is required to delete only those entries created by
+# the asic that is rebooted. Entries from the following tables are deleted in the order
+# given below
+#   (1) SYSTEM_NEIGH
+#   (2) SYSTEM_INTERFACE
+#   (3) SYSTEM_LAG_MEMBER_TABLE
+#   (4) SYSTEM_LAG_TABLE
+#   (5) The corresponding LAG IDs of the entries from SYSTEM_LAG_TABLE
+#       SYSTEM_LAG_ID_TABLE and SYSTEM_LAG_ID_SET are adjusted appropriately
+function clean_up_chassis_db_tables()
+{
+    if [[ !($($SONIC_DB_CLI CHASSIS_APP_DB PING | grep -c True) -gt 0) ]]; then
+        return
+    fi
+
+    lc=`$SONIC_DB_CLI CONFIG_DB  hget 'DEVICE_METADATA|localhost' 'hostname'`
+    asic=`$SONIC_DB_CLI CONFIG_DB  hget 'DEVICE_METADATA|localhost' 'asic_name'`
+
+    # First, delete SYSTEM_NEIGH entries
+    $SONIC_DB_CLI CHASSIS_APP_DB EVAL "
+    local host = string.gsub(ARGV[1], '%-', '%%-')
+    local dev = ARGV[2]
+    local ps = 'SYSTEM_NEIGH*|' .. host .. '|' .. dev
+    local keylist = redis.call('KEYS', 'SYSTEM_NEIGH*')
+    for j,key in ipairs(keylist) do
+        if string.match(key, ps) ~= nil then
+            redis.call('DEL', key)
+        end
+    end
+    return " 0 $lc $asic
+
+    sleep 30
+
+    # Next, delete SYSTEM_INTERFACE entries
+    $SONIC_DB_CLI CHASSIS_APP_DB EVAL "
+    local host = string.gsub(ARGV[1], '%-', '%%-')
+    local dev = ARGV[2]
+    local ps = 'SYSTEM_INTERFACE*|' .. host .. '|' .. dev
+    local keylist = redis.call('KEYS', 'SYSTEM_INTERFACE*')
+    for j,key in ipairs(keylist) do
+        if string.match(key, ps) ~= nil then
+            redis.call('DEL', key)
+        end
+    end
+    return " 0 $lc $asic
+
+    # Next, delete SYSTEM_LAG_MEMBER_TABLE entries
+    $SONIC_DB_CLI CHASSIS_APP_DB EVAL "
+    local host = string.gsub(ARGV[1], '%-', '%%-')
+    local dev = ARGV[2]
+    local ps = 'SYSTEM_LAG_MEMBER_TABLE*|' .. host .. '|' .. dev
+    local keylist = redis.call('KEYS', 'SYSTEM_LAG_MEMBER_TABLE*')
+    for j,key in ipairs(keylist) do
+        if string.match(key, ps) ~= nil then
+            redis.call('DEL', key)
+        end
+    end
+    return " 0 $lc $asic
+
+    sleep 15
```

Comment:
Same question as above... can we eliminate unnecessary sleep if it is not really needed...

Reply:
This is for the similar reason as explained for the previous comment but for SYSTEM_LAG_TABLE and SYSTEM_LAG_MEMBER_TABLE entries. At the line card/asic, deletion of system lag will  happen only if all the members of the system lag are removed. Even though we send  system lag member delete before sending the system lag delete, due to the order of processing and other dependency (like system inteface delete and neighbours delete which introduces timing differences), by the time system lag is tried to be deleted, sometimes, the a system lag still has system interface pending to be deleted. Having this delay (the lowest rounded time we tested with which we did not see the issue) helps to make sure that all system lag members are deleted before deleting system lag

Reply:
Thanks for the explanation.  Please add a comment to disclose this so that in future someone may not try to remove it or at least when try to better tunning the value will know what to verify so it will not end up breaking something...

Reply:
Added comment